hbase-client/src/main/java/org/apache/hadoop/hbase/client/RpcRetryingCallerImpl.java

```diff
@@ -102,7 +103,13 @@ public T callWithRetries(RetryingCallable<T> callable, int callTimeout)
       long expectedSleep;
       try {
         // bad cache entries are cleared in the call to RetryingCallable#throwable() in catch block
-        callable.prepare(tries != 0);
+        Throwable t = null;
+        if (exceptions != null && !exceptions.isEmpty()) {
+          t = exceptions.get(exceptions.size() - 1).throwable;
+        }
+        if (!(t instanceof RpcThrottlingException)) {
+          callable.prepare(tries != 0);
+        }
```

Comment:
So, the idea here is that if we got an exception and we are retrying, do NOT reload cache if the exception was a because we were throttled?

This is a good idea. I wonder if there are other exceptions where we retry but do not need to reload the cache?